llvm/lib/Transforms/IPO/FunctionImport.cpp

```diff
@@ -174,6 +174,10 @@ static cl::opt<std::string> WorkloadDefinitions(
              "}"),
     cl::Hidden);
 
+static cl::opt<std::string>
+    ContextualProfile("thinlto-pgo-ctx-prof",
```

Comment:
Can you add a test to exercise this handling?

Reply:
We don't currently have a textual representation of the contextual profile, and the import manager is internal to this file, meaning we can't write a unittest. I'd do either, though, if the new functionality was more involved, but it's really just taking data from the ctx profile reader - which is tested - and filling a set with it.

I refactored a bit more to leave the specific code to a minimum.

Reply:
I meant a lit test. I think it would be good to add (reading the binary format like I am assuming the ctx profile reader tests do?), otherwise this is untested functionality.

Reply:
Right - need to figure out where to place it, because it would need to be a test that runs a program to generate a profile. Meaning it needs compiler-rt.

Reply:
done.